Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerTreeView.cpp

```diff
@@ -108,61 +91,109 @@ namespace AzToolsFramework
             }
         }
     }
 
-    void EntityOutlinerTreeView::checkExpandedState(const QModelIndex& current)
+    void EntityOutlinerTreeView::CheckExpandedState(const QModelIndex& current)
     {
         const bool expandState = current.data(EntityOutlinerListModel::ExpandedRole).template value<bool>();
         setExpanded(current, expandState);
     }
 
     void EntityOutlinerTreeView::mousePressEvent(QMouseEvent* event)
     {
-        //postponing normal mouse pressed logic until mouse is released or dragged
-        //this means selection occurs on mouse released now
-        //this is to support drag/drop of non-selected items
+        // Postponing normal mouse pressed logic until mouse is released or dragged.
+        // This allows drag/drop of non-selected items.
         ClearQueuedMouseEvent();
         m_queuedMouseEvent = new QMouseEvent(*event);
     }
 
-    void EntityOutlinerTreeView::mouseReleaseEvent(QMouseEvent* event)
+    void EntityOutlinerTreeView::mouseMoveEvent(QMouseEvent* event)
     {
-        if (m_queuedMouseEvent && !m_draggingUnselectedItem)
+        bool forceUpdate = false;
+
+        QModelIndex previousHoveredIndex = m_currentHoveredIndex;
+        m_mousePosition = event->pos();
+
+        if (QModelIndex hoveredIndex = indexAt(m_mousePosition);
+            hoveredIndex.column() != EntityOutlinerListModel::ColumnSpacing && m_currentHoveredIndex != hoveredIndex)
+        {
+            m_currentHoveredIndex = hoveredIndex;
+        }
+        else
         {
-            // mouseMoveEvent will set the state to be DraggingState, which will make Qt ignore 
-            // mousePressEvent in QTreeViewPrivate::expandOrCollapseItemAtPos. So we manually 
-            // and temporarily set it to EditingState.
-            QAbstractItemView::State stateBefore = QAbstractItemView::state();
-            QAbstractItemView::setState(QAbstractItemView::State::EditingState);
+            m_currentHoveredIndex = QModelIndex();
+        }
 
-            //treat this as a mouse pressed event to process selection etc
-            processQueuedMousePressedEvent(m_queuedMouseEvent);
+        if (m_queuedMouseEvent)
+        {
+            if (!m_isDragSelectActive)
+            {
+                QModelIndex clickedIndex = indexAt(m_queuedMouseEvent->pos());
+                if (clickedIndex.isValid() && clickedIndex.column() != EntityOutlinerListModel::ColumnSpacing)
+                {
+                    HandleDrag();
+                }
+                else
+                {
+                    m_isDragSelectActive = true;
+                    forceUpdate = true;
+                }
+            }
+            else
+            {
+                SelectAllEntitiesInSelectionRect();
+                forceUpdate = true;
+            }
+        }
 
-            QAbstractItemView::setState(stateBefore);
+        if (previousHoveredIndex != m_currentHoveredIndex)
+        {
+            forceUpdate = true;
+        }
```

Comment:
uber nit (can ignore):

This could be `forceUpdate = previousHoveredIndex != m_currentHoveredIndex || forceUpdate` (just a random style thing, other way might be clearer to others)